torchaudio/io/_playback.py

```diff
@@ -0,0 +1,76 @@
+from sys import platform
+from typing import Optional
+
+import torch
+import torchaudio
+from torchaudio.io import StreamWriter
+
+dict_format = {
+    torch.uint8: "u8",
+    torch.int16: "s16",
+    torch.int32: "s32",
+    torch.int64: "s64",
+    torch.float32: "flt",
+    torch.float64: "dbl",
+}
+
+
+def play_audio(
+    waveform: torch.Tensor,
+    sample_rate: Optional[float],
+    device: Optional[str] = None,
+) -> None:
+    """Play audio through specified or available output device. This
+    function is only available on MacOS.
+
+    Args:
+        waveform: Tensor containing the audio to play.
+            Expected shape: (time) for mono or (time, channels) for stereo signals.
+        sample_rate: Sample rate of the input audio
+        device: Output device to use. If None, the default device is used.
+    """
+
+    if waveform.dtype in dict_format.keys():
+        format = dict_format[waveform.dtype]
+    else:
+        raise ValueError(
+            f"Unsupported dtype: {waveform.dtype}. \
+            The list of supported dtypes is: {dict_format.keys()}"
+        )
+
+    if waveform.ndim > 2:
+        raise ValueError(
+            f"Expected 2D stereo tensor with shape \
+            (time, channels), or 1D mono tensor with shape (time), \
+                got {waveform.ndim}D tensor instead"
+        )
+    elif waveform.ndim == 1:
+        waveform = waveform.unsqueeze(-1)
+
+    if platform == "darwin":
+        device = device or "audiotoolbox"
+        path = "-"
+    else:
+        raise ValueError(
+            f"This function only supports MacOS, \
+            but current OS is {platform}"
+        )
+
+    available_devices = list(torchaudio.utils.ffmpeg_utils.get_output_devices().keys())
+    if device not in available_devices:
+        raise ValueError(
+            f"Device {device} is not available. \
+            Available devices are: {available_devices}"
```

Comment:
Please do not use backslash at all.
The use of backslash like this make the indentation part of the message.

```python
>>> platform="foo"
>>> print(
...             f"This function only supports MacOS, \
...             but current OS is {platform}"
...         )
This function only supports MacOS,             but current OS is foo
```

```suggestion
            f"Device {device} is not available. Available devices are: {available_devices}"
```